libethereum/EthereumPeer.cpp

```diff
@@ -88,8 +86,7 @@ void EthereumPeer::setIdle()
 
 void EthereumPeer::requestStatus()
 {
-	if (m_asking != Asking::Nothing)
-	clog(NetWarn) << "Bad state: requesting state should be the first action";
+	assert(m_asking == Asking::Nothing);
```

Comment:
this is /very/ harsh; why bother crashing the app in the case it's wrong rather than attempting to recover?